resources/js/views/Collection.vue

```diff
@@ -143,22 +143,12 @@ const removeCollection = () => {
     itemsStore.removeAll()
 }
 
-const shareCollection = () => {
+const shareCollection = async () => {
     loading.value = true
-    axios
-        .post('/api/collections', {
-            items: itemsStore.items,
-        })
-        .then((res) => {
-            shareUrl.value = res.data.url
-            shareUrlDialog()
-        })
-        .catch((err) => {
-            console.log(err)
-        })
-        .finally(() => {
-            loading.value = false
-        })
+    const value = await itemsStore.getCollectionLink()
+    shareUrl.value = value
```

Comment:
any reason why not directly
```suggestion
    shareUrl.value = await itemsStore.getCollectionLink()
```
?